drivers/console/aarch64/console.S

```diff
@@ -95,10 +95,15 @@ endfunc console_getc
 	 * data that hasn't been output. It returns 0
 	 * upon successful completion, otherwise it
 	 * returns -1.
-	 * Clobber list : x0, x1
+	 * Clobber list : x0, x1, x2, x3, x4
 	 * ---------------------------------------------
 	 */
 func console_flush
+#if COREBOOT
+	mov	x4, x30
+	bl	coreboot_cbmc_flush
+	mov	x30, x4
+#endif
```

Comment:
Okay, how about this: I add a new series of assembler macros like

 .macro func_console_putc _name
 func \_name
 .weak console_single_putc
 console_single_putc:
 .endm

Then I rewrite all the console drivers to use this (with a unique name) instead of the existing 'func console_core_putc' (and equivalent for the other functions). I would change drivers/console/.../console.S to use structs with function pointers, with a new function to register them (e.g. console_register). It would still provide console_init for backwards compatibility, which would call console_register with a struct containing console_core_putc and friends as function pointers.

This way, existing platforms that only link a single console driver and call console_init should continue to work unchanged. Newer platforms could start linking more than one driver and use console_register instead (so that console_init is garbage-collected by the linker and the duplicate .weak symbols aren't referenced so that they shouldn't cause a problem).

Does that sound like the kind of system you want?